src/Symfony/Component/Console/Helper/DialogHelper.php

```diff
@@ -190,7 +190,7 @@ public function ask(OutputInterface $output, $question, $default = null, array $
 
                     foreach ($autocomplete as $value) {
                         // If typed characters match the beginning chunk of value (e.g. [AcmeDe]moBundle)
-                        if (0 === strpos($value, $ret) && $i !== strlen($value)) {
+                        if (0 === strpos($value, $ret) && !isset($value[$i-1])) {
```

Comment:
this is not equivalent to the previous code. You don't handle shorter strings the same anymore. Your new code is checking `` strlen($value) < $i - 1``